File: livingrimoire start here/LivinGrimoire java/src/LivinGrimoire/Cerabellum.java
package LivinGrimoire;

public class Cerabellum {
    // runs an algorithm
    private int fin;
    private int at;
    private Boolean incrementAt = false;

    public void advanceInAlg() {
        if (incrementAt) {
            incrementAt = false;
            at++;
            if (at == fin) {
                isActive = false;
            }
        }
    }

    public int getAt() {
        return at;
    }

    public Algorithm alg;
    private boolean isActive = false;
    private String emot = "";

    public String getEmot() {
        return emot;
    }
    public boolean setAlgorithm(Algorithm algorithm) {
        if (!isActive && (!algorithm.getAlgParts().isEmpty())) {
            this.alg = algorithm;
            this.at = 0;
            this.fin = algorithm.getSize();
            this.isActive = true;
            this.emot = alg.getAlgParts().get(at).myName(); // updated line
            return false;
        }
        return true;
    }

    public boolean isActive() {
        return isActive;
    }

    public String act(String ear, String skin, String eye) {
        String axnStr = "";
        if (!isActive) {
            return axnStr;
        }
        if (at < fin) {
            axnStr = alg.getAlgParts().get(at).action(ear, skin, eye);
            this.emot = alg.getAlgParts().get(at).myName();
            if (alg.getAlgParts().get(at).completed()) {
                incrementAt = true;
            }
        }
        return axnStr;
    }
    public void deActivation(){
        this.isActive = this.isActive && !alg.getAlgParts().get(at).algKillSwitch;
    }
}

File: livingrimoire start here/LivinGrimoire Kotlin/kotlin/livinGrimoire/Cerabellum.kt
package livinGrimoire

class Cerabellum {
    // runs an algorithm
    private var fin = 0
    var at = 0
        private set
    private var incrementAt = false
    fun advanceInAlg() {
        if (incrementAt) {
            incrementAt = false
            at++
            if (at == fin) {
                isActive = false
            }
        }
    }

    var alg: Algorithm? = null
    var isActive = false
        private set
    var emot = ""
        private set

    fun setAlgorithm(algorithm: Algorithm): Boolean {
        if (!isActive && !algorithm.algParts.isEmpty()) {
            alg = algorithm
            at = 0
            fin = algorithm.size
            isActive = true
            emot = alg!!.algParts.get(at).myName() // updated line
            return false
        }
        return true
    }

    fun act(ear: String, skin: String, eye: String): String {
        var axnStr = ""
        if (!isActive) {
            return axnStr
        }
        if (at < fin) {
            axnStr = alg!!.algParts.get(at).action(ear, skin, eye)
            emot = alg!!.algParts.get(at).myName()
            if (alg!!.algParts.get(at).completed()) {
                incrementAt = true
            }
        }
        return axnStr
    }

    fun deActivation() {
        isActive = isActive && !alg!!.algParts.get(at).algKillSwitch
    }
}